sonaranalyzer-dotnet/src/SonarAnalyzer.Common/Rules/Hotspots/DoNotHardcodeCredentialsBase.cs

```diff
@@ -148,13 +151,30 @@ private IEnumerable<string> FindCredentialWords(string variableName, string vari
                 var matches = analyzer.passwordValuePattern.Matches(variableValue);
                 foreach (Match match in matches)
                 {
-                    credentialWordsFound.Add(match.Groups["password"].Value);
+                    if (!match.Success)
```

Comment:
`if(!Success) continue` followed by single if statement in the loop seems difficult to understand from cyclomatic complexity POV. I suggest to change it to
`if (match.Success && !IsValidParameter(match.Groups["suffix"].Value))`